Specifications/QIR/Callables.md

```diff
@@ -318,6 +312,12 @@ language-specific compiler.
 The lambda body may need to include additional code that performs argument tuple
 construction before calling the underlying callable.
 
+#### Memory Management Table
+
+Any captured values need to remain alive as long as the callable value exists, and correspondingly may also need to be unreferenced when the type information for the captured values cannot be statically determined. Upon creation, a table with two function pointers is hence associated with a callable value. 
```

Comment:
Maybe, explain here that the memory management is recursive? (Otherwise, the callable could manage the capture tuple automatically and it's not clear why these additional facilities are needed.)

Reply:
I expanded a bit more here.